pkg/adapter/cronjobevents/adapter.go

```diff
@@ -49,6 +49,22 @@ type Adapter struct {
 	client client.Client
 }
 
+// Initialize cloudevent client
+func (a *Adapter) initClient() error {
+	if a.client == nil {
+		var err error
+		if a.client, err = client.NewHTTPClient(
```

Comment:
In this case, I think it would be more fluent to do the opposite:

```golang
a.client, err := client.NewHTTPClient(
      ....
      ....)
if err != nil {
    return err
}
```

Two rubrics for choosing between the two forms:

1. Avoid the inline `if` assignment if the function call is multiple lines; it's harder to follow the `if` block that way.
2. Which is fewer lines of code? In this case, needing to pre-declare `var err error` in order to be able to assign to `a.client` removes all the benefit of inlining the assignment in the `if` statement.

Reply:
@evankanderson - current golang doesn't support `a.client, err := client.NewHTTPClient....`

see the issue: 
https://github.com/golang/go/issues/30318

Reply:
and this:  https://github.com/golang/go/issues/6842